src/main/java/reposense/git/GitRevList.java

```diff
@@ -57,15 +57,28 @@ public static String getCommitHashInRange(String root, String branchName, String
             return startHash;
         }
 
-        Path rootPath = Paths.get(root);
-        String revListCommand = "git rev-list " + startHash + ".." + endHash + " " + branchName
-                + REVISION_PATH_SEPARATOR;
-        String output = runCommand(rootPath, revListCommand);
+        String fromStartHash = getAllCommitHashSince(root, branchName, startHash);
+        String fromEndHash = getAllCommitHashSince(root, branchName, endHash);
+        StringBuilder output = new StringBuilder();
 
-        if (output.equals("")) {
-            return endHash + "\n" + startHash;
+        if (fromStartHash.length() > fromEndHash.length()) {
```

Comment:
I understand that this segment of code is trying to find which hash is earlier, might need to refactor or place a comment here as it is slightly unclear and hard to understand at first glance